ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +405,186 @@ def test_trustdomain_disable_disables_subdomain(self):
+                tmpdir, 'ssh_priv_key'), 'w') as fp:
+            fp.write(privkey)
+
+        keys_opts = ' '.join(['--ssh "{}"'.format(k) for k in pub_keys])
+        cmd = 'ipa user-mod {} {}'.format(username, keys_opts)
+        master.run_command(cmd)
+
+        # connect with first SSH key
+        user_ssh_priv_key = os.path.join(tmpdir, 'ssh_priv_key')
+
+        ssh_key_file_contents = ''
+        with open(user_ssh_priv_key, 'r') as infile:
+            ssh_key_file_contents = infile.read()
+        master.put_file_contents('/tmp/user_ssh_priv_key',
+                                 ssh_key_file_contents)
+        master.run_command(['chmod', '600', '/tmp/user_ssh_priv_key'])
+
+        # add group a
+        cmd_output = master.run_command(['ipa', 'group-add', 'a'])
+        assert 'Added group "a"' in cmd_output.stdout_text
+
+        # add group b
+        cmd_output = master.run_command(['ipa', 'group-add', 'b'])
+        assert 'Added group "b"' in cmd_output.stdout_text
+
+        # add group c
+        cmd_output = master.run_command(['ipa', 'group-add', 'c'])
+        assert 'Added group "c"' in cmd_output.stdout_text
+
+        # add group members
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'b', '--groups', 'a'])
+        assert 'Group name: b' in cmd_output.stdout_text
+        assert 'Member groups: a' in cmd_output.stdout_text
+        assert 'Number of members added 1' in cmd_output.stdout_text
+
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'c', '--groups', 'b'])
+        assert 'Group name: c' in cmd_output.stdout_text
+        assert 'Member groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member groups: a' in cmd_output.stdout_text
+
+        # add user to group 'a'
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'a', '--users', username])
+        assert 'Group name: a' in cmd_output.stdout_text
+        assert 'Member users: {}'.format(username) in cmd_output.stdout_text
+        assert 'Member of groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member of group: c' in cmd_output.stdout_text
+
+        # clear sssd_cache
+        clear_sssd_cache(master)
+
+        # user lookup
+        # at this point, testuser001 has the following group memberships
+        # Member of groups: a, ipausers
+        # Indirect Member of group: b, c
+        cmd_output = master.run_command(['ipa', 'user-show', username])
+        assert 'a' in cmd_output.stdout_text
+        assert 'ipausers' in cmd_output.stdout_text
+        assert 'b' in cmd_output.stdout_text
+        assert 'c' in cmd_output.stdout_text
+
+        clear_sssd_cache(client)
+
+        try:
+            cmd = ['ssh', '-i', '/tmp/user_ssh_priv_key',
+                   '-q', '{}@{}'.format(username, client.hostname)]
```

Comment:
You need to add a command at the end of ssh, for instance:
```
            cmd = ['ssh', '-i', '/tmp/user_ssh_priv_key',
                   '-q', '{}@{}'.format(username, client.hostname), 'true']
```
otherwise the ssh command will not return. Same comment for all the places calling ssh.

Reply:
thanks for pointing it out